### ISSUE TITLE
Cache latents slow down after a certain number of steps have passed

### ISSUE THREAD

**Issue by aearone** [bmaltais/kohya_ss#1209]
I don't know what happened, it's never happened before. I am trying to train the model at 576x576 resolution using regularization images (I have 1960 of them and 98 training images), but when I check the cache after 42 steps the speed drops a lot, i.e. it was 5-8s/it and becomes 1.4, cache read time is 1 hour. What can be the problem? I checked the images, they are fine, they all have the same format and size.

![Screenshot_50](https://github.com/bmaltais/kohya_ss/assets/93353660/ff46b47e-bc24-478e-ae18-9a33da0f0beb)

**Comment by bmaltais** [comment]
You may want to report to Kohya in his repo and let him know that this is related to his new SDXL code... In the meantime you can revert back to a previous release that worked before the sdxl code release.

**Comment by aearone** [comment]
> You may want to report to Kohya in his repo and let him know that this is related to his new SDXL code... In the meantime you can revert back to a previous release that worked before the sdxl code release.

Thank you, that's what I did. I also found some regularity between training resolution and regularization images resolution, i.e. if training resolution is 576x576 and regularization images resolution is 512x512, there is no decrease in cache read rate, but if training resolution and regularization images resolution are the same - there is a decrease in cache read rate. And the higher the resolution, the slower it happens. But I have never understood for what reason there is a sharp decrease in the read rate after a certain number of steps. This is obviously a problem on the script side.

**Comment by aearone** [comment]
> You may want to report to Kohya in his repo and let him know that this is related to his new SDXL code... In the meantime you can revert back to a previous release that worked before the sdxl code release.

And so, I did a complete reinstallation of the Windows system and checked to see if the problem would remain. Just as I thought, same problem on a completely new operating system. And as I found out, the reason is the lack of VRAM and RAM, as all 16gb RAM and 8 VRAM are used 100% when caching. If you close all third-party programs and applications caching speed increases, there is a direct correlation. So I think that the problem is only on the system side. Maybe it will be useful for someone to know the reason. I'm closing this topic.
